src/tools/Cargo.toml

```diff
@@ -0,0 +1,12 @@
+[package]
```

Comment:
You said:
> Having a single src folder for one package, but then another package inside of it (s3)

So, if you're confused by putting all the crates in `src`, shouldn't `tools` be at the root? And I'd avoid `src/tools/src/bin/*.rs` (notice two `src`s!).

I imagine:
```
.
├── s3 (if you want, but it integrates `cargo-registry` so I think it's better to be in `src`)
├── src (a.k.a. `cargo-registry`)
└── tools (`src/bin` previously. I guess `utils` is better? `src/bin` is a common structure, but if we have multiple crates, it might be better to be at root for clarify.)
	└── on_call(if you want to separate it. It should be under `src/bin` since it's only for bin utils.)
```

Reply:
we can also move them to the root folder, but I wanted to avoid cluttering the root folder with all those package folders for now.

Reply:
Then, it may be good to have `crates/` dir, like:
```
.
└── crates
	├── cargo-registry
	├	└── s3
	└── tools (or `utils`)
		└── on_call
```